### ISSUE TITLE
Pre-launch checklist in the dashboard + Live toggle

### ISSUE THREAD

**Issue by DrewAPicture** [SellBirdHQ/theme#30]
With the work being done in #18 to add Stripe Connect-ivity, I think we should look at how we can convey steps that should be taken before taking a store "live", including the Stripe connection, maybe adding products, configuring specific settings (if needed), and then some kind of toggle for live vs [no live].

**Comment by pippinsplugins** [comment]
I'd really love to see this as a nice set of on-boarding steps that give the user a clear visual of steps they need to complete.

**Comment by amdrew** [comment]
👍 I'll work on gathering some inspiration so we can start to think how this will look/work.

**Comment by DrewAPicture** [comment]
Might be worth looking at whether we could use the jquery-steps plugin here.

**Comment by amdrew** [comment]
That's one idea for sure. Another is a simple vertically stacked list of items to complete with an overall progress bar. Might be easier to see what needs to be done and the user wouldn't need to launch any kind of stepped progress mechanism to get to that 4th step they missed the first time round.

**Comment by amdrew** [comment]
A combination of both could work nicely. A stepped set up window for the initial process. If a customer skips any step and doesn't completely finish, we can record that step(s) and then show it to them later.

**Comment by amdrew** [comment]
Posted some research to Trello. Based on what I've seen so far, I think we should keep it simple and provide a list of items on the dashboard.

**Comment by DrewAPicture** [comment]
@amdrew Worth mentioning that we kind of need to divorce the idea of a member being active from a store being live. This is because we can't do things like create pages, save settings, connect to Stripe, or any number of other operations unless the site exists – which is only true if the member is active.

So we may need to look at modifications to the store theme based on a new site option to reflect whether a store is "live" or not, if that makes sense.

**Comment by amdrew** [comment]
@DrewAPicture Can you elaborate a little further? When a user completes the registration form, a site is created, and we present them with a list of items that need to be completed before the site is deemed "live". What modifications to the store theme do you envision being needed based on whether the site is live or not?

**Comment by DrewAPicture** [comment]
I meant in terms of disallowing access to the store (or at least substantive parts parts of the store) at the template level based on something other than the membership status. Though to be fair, it feels a little bit like we're working _around_ a limitation in RCP/RCP site creation more than anything.

e.g. (1) The membership has to be active (2) for the site to be created (3) for us to store options in the site.

I was talking to @pippinsplugins about this a little bit in Slack, and wonder if simply putting a filter on the required status for creating a site might not be the best way forward. That way, we could actually use and rely on the membership status to set the tone of "is this store live or not?" while simultaneously having the site always be created, and then ultimately be able to control "access" via RCP at the template level which is what I think we want in the first place.

**Comment by DrewAPicture** [comment]
I don't know what the solution is if we're planning to use all of our products (EDD, RCP, RCP Site Creation, AffWP et al) _exactly as intended_. Ideally you could have an RCP status that signified paid but not live, which would allow us to leverage RCP's built-in access control APIs for the "active" status when the store is actually converted to "live". Do you see the problem?

**Comment by pippinsplugins** [comment]
I think we should create a custom indicator for "store is ready". This could be a helper function that looks at the following conditions:

1. Membership is active
2. Stripe Connect has been connected
3. Site exists
4. All necessary pages for site exists

So long as all four conditions return `true`, we consider the store is ready.

We could also have a method of returning an array of key=>value pairs that provides a `true|false` value for each of the four items above.

**Comment by amdrew** [comment]
In addition to the above, there should be a final "go live" toggle on the dashboard, once the above conditions are met.

**Comment by DrewAPicture** [comment]
Almost like a checklist :-P

**Comment by DrewAPicture** [comment]
Closing in favor of #280 per today's scrub.

Registration -> intermediate onboarding form -> Dashboard

**Comment by amdrew** [comment]
We'll need to keep this in addition to #280. While the intermediate on-boarding form is required (i.e collecting their name, selecting what they intend to sell) this could be a wizard or to-do list from within the dashboard. Tasks or helpful to-do reminders that could be completed, or just dismissed altogether.
